chapters/planning-data-work.tex

```diff
@@ -151,162 +469,185 @@ \subsection{Code management}
-but information that is relevant to understand the code should always be written in the code itself.
+not just a means to an end,
+and should be written thinking of how easy it will be for someone to read it later.
 
-Code organization is the next level.
-Start by adding a code header.
-This should include simple things such as stating the purpose of the script and the name of the person who wrote it.
+Code documentation is one of the main factors that contribute to readability.
+Start by adding a code header to every file.
+This should include simple things such as the purpose of the script and the name of the person who wrote it.
 If you are using a version control software,
 the last time a modification was made and the person who made it will be recorded by that software.
 Otherwise, you should include it in the header.
-Finally, and more importantly, use it to track the inputs and outputs of the script.
-When you are trying to track down which code creates a data set, this will be very helpful.
-
-Breaking your code into readable steps is also good practice on code organization.
+Finally, use the header to track the inputs and outputs of the script.
+When you are trying to track down which code creates which data set, this will be very helpful.
+While there are other ways to document decisions related to creating code
+(GitHub offers a lot of different documentation options, for example),
+the information that is relevant to understand the code should always be written in the code file.
+
+In the script, alongside the code, are two types of comments that should be included.
+The first type of comment describes what is being done.
+This might be easy to understand from the code itself
+if you know the language well enough and the code is clear,
+but often it is still a great deal of work to reverse-engineer the code's intent.
+Writing the task in plain English (or whichever language you communicate with your team on)
+will make it easier for everyone to read and understand the code's purpose
+-- and also for you to think about your code as you write it.
+The second type of comment explains why the code is performing a task in a particular way.
+As you are writing code, you are making a series of decisions that
+(hopefully) make perfect sense to you at the time.
+These are often highly specialized and may exploit a functionality
+that is not obvious or has not been seen by others before.
+Even you will probably not remember the exact choices that were made in a couple of weeks.
+Therefore, you must document your precise processes in your code.
+
+Code organization means keeping each piece of code in an easily findable location.
+  \index{code organization}
+Breaking your code into independently readable ``chunks'' is one good practice on code organization,
+because it ensures each component does not depend on a complex program state
+created by other chunks that are not obvious from the immediate context.
```

Comment:
Just breaking the code into chunks does not "ensures each component does not depend on a complex program state
created by other chunks that are not obvious from the immediate context"

You need to write that code in a modular fashion for that to be true, and that is not what this paragraph is saying. Either we should reduce what we promise, or write something about these _chunks_ must be written independent on each other. You are saying that is a result of chunkification, but I think that is a requirement of chunkification having any benefit.

Reply:
80a42c84df22acb662032c394e36305777a479b0